### PR TITLE
Update to newest Reason & React versions

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Reason React has changed some of its syntax and library recently, such as replacing `ReasonReact.stringToElement` with `ReasonReact.string`.

This PR updates bs-platform, reason-react, reason, and reason-scripts to use the newest versions + replaces some of the syntax.

P.S. Thanks again for creating this binding!

### CODE REVIEW

File: examples/src/Example1.re [hunk 1]
```diff
@@ -24,15 +24,13 @@ let make = (~options, _children) => {
...
-          self.state.selectedItem,
-        )
-      )
-      onChange=(selected => self.send(Change(selected)))
-      arrowRenderer=((_) => <div> (ReasonReact.stringToElement("+")) </div>)
-      filterOptions=(Func((~options, ~filter as _filter) => options))
-      placeholder=(Str("Select something.."))
+      value=?{
+        self.state.selectedItem
+        |> Js.Option.map((. b) => Select.Option.Val(b))
```
Comment (Eric-Arellano) [inline examples/src/Example1.re]: Reason community seems to prefer `Belt.Option` now. I spent a minute trying to change to this, but couldn't figure it out so moved on.

Comment (liubko) [reply examples/src/Example1.re]: what is the error is?

Generally speaking smth like this should work
```
open Belt;
...
Option.map(self.state.selectedItem, item => Select.Option.Val(item))
```

**Comment by Eric-Arellano** [comment]
I'm getting these two exceptions. 

screen shot 2018-09-19 at 7 06 17 am

screen shot 2018-09-19 at 7 06 24 am

I'm confused because I don't get these exceptions when using essentially the same code as this example on my personal project? I'm using the most recent version of this library, which I assumed meant this is not an issue with the bindings themselves.

--
Thanks for the Belt.Option snippet! That's much better and I now finally understand `->` vs `|>` 😂

**Comment by Eric-Arellano** [comment]
Hm, bumping React and ReactDOM to 16.4.2 didn't fix it. 

Let me know if anything pops out at you. I'm trying to inspect the generated JS to see where the differences between my personal project are.

**Comment by Eric-Arellano** [comment]
Got it to work! The issue was the library using the old version of Reason and BS Platform. Bumping those fixed everything.

screen shot 2018-09-19 at 7 43 28 am

This is ready to merge after a code review. Note that it may break people still using old versions of Reason and BS Platform, so I bumped the version numbers.

File: examples/package.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "@ahrefs/bs-react-select-examples",
-  "version": "0.0.1",
+  "version": "0.0.2",
```
Comment (Eric-Arellano) [inline examples/package.json:3]: Should this version correspond with the library's version?

Comment (liubko) [reply examples/package.json]: I don't think it's necessary, this can go out of sync really easily. for example when patching any of libraries

Comment (liubko) [reply examples/package.json:3]: this is just an example, I don't think we need to change a version here at all.

**Comment by liubko** [comment]
@Eric-Arellano thanks for a contribution. Merged and published new version

**Comment by Eric-Arellano** [comment]
Awesome! Thank you for helping me along the way.

**Pull request merged**